File: Memory-JAVA/app/src/main/java/com/andraganoid/memory_java/GridMemoryAdapter.java
package com.andraganoid.memory_java;

import android.content.Context;
import android.util.TypedValue;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.GridView;
import android.widget.ImageView;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;

public class GridMemoryAdapter extends BaseAdapter {

    Context context;
    List <Field> fList = new ArrayList <>();
    LayoutInflater inflater;
    TextView item;
    ImageView back;
    int itemHeight;
    Field currentField;

    public GridMemoryAdapter(Context context, int itemHeight) {
        this.context = context;
        inflater = LayoutInflater.from(context);
        this.itemHeight = itemHeight;
        setFields(null);
    }

    @Override
    public int getCount() {
        return fList.size();
    }

    @Override
    public Object getItem(int position) {
        return null;
    }

    @Override
    public long getItemId(int position) {
        return 0;
    }

    @Override
    public View getView(int position, View view, ViewGroup parent) {
        view = inflater.inflate(R.layout.field_row, parent, false);
        view.setLayoutParams(new GridView.LayoutParams(GridView.AUTO_FIT, itemHeight));
        item = view.findViewById(R.id.field_item);
        back = view.findViewById(R.id.field_back_img);
        item.setTextSize(TypedValue.COMPLEX_UNIT_PX, itemHeight / 2);

        currentField = fList.get(position);
        item.setText(String.valueOf(currentField.getItem()));
        if (currentField.isSolved()) {
            item.setBackgroundColor(context.getResources().getColor(R.color.colorAccent));
        }

        if (currentField.isOpen() || currentField.isSolved()) {
            item.setVisibility(View.VISIBLE);
            back.setVisibility(View.GONE);
        } else {
            item.setVisibility(View.GONE);
            back.setVisibility(View.VISIBLE);
        }

        return view;
    }

    public void setFields(List <Field> lf) {
        if (lf != null) {
            fList = lf;
        } else {
            fList.clear();
        }
        notifyDataSetChanged();
    }
}

File: Memory-KOTLIN/app/src/main/java/com/andraganoid/memory_kotlin/GridMemoryAdapter.kt
package com.andraganoid.memory_kotlin

import android.content.Context
import android.graphics.Color
import android.util.TypedValue
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.*
import androidx.core.content.ContextCompat
import kotlinx.android.synthetic.main.field_row.view.*
import java.util.ArrayList



class GridMemoryAdapter(internal val context: Context, internal val itemHeight: Int) : BaseAdapter() {
    internal var fList: List<Field> = listOf()
  //  internal var inflater: LayoutInflater
//    internal var item: TextView
//    internal var back: ImageView
//    internal var currentField: Field

    init {

       // inflater = LayoutInflater.from(context)
       setFields(fList)
    }

    override fun getCount(): Int {

       //  return  fList?.size ?: 0;
        return fList.size
    }

    override fun getItem(position: Int): Any? {
        return null
    }

    override fun getItemId(position: Int): Long {
        return 0
    }

    override fun getView(position: Int, view: View?, parent: ViewGroup?): View {




//
//
//
  //  var
//
//      //  var inflater :LayoutInflater
//        var inflator = context!!.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
//        var view = inflator.inflate(R.layout.field_row, null)
//
          val  gView  = LayoutInflater.from(context).inflate(R.layout.field_row, parent, false)
        gView.layoutParams = AbsListView.LayoutParams(GridView.AUTO_FIT, itemHeight)
        val item = gView.fieldItem;
        val back = gView.fieldBackImg;
        item.setTextSize(TypedValue.COMPLEX_UNIT_PX, (itemHeight / 2).toFloat())

        val currentField = fList[position]
        item.setText(currentField.item.toString())
        if (currentField.isSolved) {
            item.setBackgroundColor(ContextCompat.getColor(context, R.color.colorAccent));
        }

        if (currentField.isOpen || currentField.isSolved) {
            item.visibility = View.VISIBLE
            back.visibility = View.GONE
        } else {
            item.visibility = View.GONE
            back.visibility = View.VISIBLE
        }

        return gView
    }

    fun setFields(lf: List<Field>) {
//        fList = if (lf != null) lf else {
//            listOf()
//        }

        fList=lf;
notifyDataSetChanged()

//        if (lf != null) {
//            fList = lf
//        } else {
//            null
//        }
    }
}